e2e/testcases/postsync_logging_test.go

```diff
@@ -320,37 +319,36 @@ func TestCloudLoggingStatusEventRepoSync(t *testing.T) {
 		nt.ClusterName,
 		startTime.Format(time.RFC3339),
 	)
-	nt.Must(waitForLogEntryInCloudLogs(nt, filter, 1, commit, repoSyncID.Name, repoSyncID.Namespace, []string{msg, "\"generation\":1"}, 120*time.Second))
+	nt.Must(waitForLogEntryInCloudLogs(nt, filter, commit, repoSyncID.Name, repoSyncID.Namespace, []string{msg, "\"generation\":1"}, 120*time.Second))
 }
 
-// assertLogEntryHasCount checks if the specified messages appear in the logs
-// exactly the expected number of times, matching the given commit, name, and namespace.
-func assertLogEntryHasCount(logs []string, expectedCount int, commit, rname, rnamespace string, messages []string) error {
-	count := 0
-
+// assertOneLogLinePerMessage checks if each specified message appears in exactly one line
+// among the log entries which match the given commit, resource name, and namespace.
+func assertOneLogLinePerMessage(logs []string, commit, resourceName, resourceNamespace string, messages []string) error {
```

Comment:
Looking at the usages of this, I think this function can be simplified to just verifying a single message instead of an array of messages (e.g. `assertLogsContainMessageOnce`). I'm thinking for each of the existing assertions that pass in more than one message, we can just validate the first message in the array (for each assertion the first message seems to be the most meaningful). This should make the function easier to understand without weakening the assertion much IMO.

WDYT?

BTW this function definitely looks like it needed a refactor - good catch 👍

Reply:
Sounds good, simplifies it a lot. For the one test where the other message seemed relevant I added a second assertion